src/applications/personalization/profile-2/components/AccountSecurityContent.jsx

```diff
@@ -0,0 +1,106 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+
+import AlertBox from '@department-of-veterans-affairs/formation-react/AlertBox';
+
+import recordEvent from 'platform/monitoring/record-event';
+import {
+  isLOA3 as isLOA3Selector,
+  isMultifactorEnabled as isMultifactorEnabledSelector,
+  selectProfile,
+} from 'platform/user/selectors';
+import { ssoe as ssoeSelector } from 'platform/user/authentication/selectors';
+
+import ProfileInfoTable from './ProfileInfoTable';
+import TwoFactorAuthorizationStatus from './TwoFactorAuthorizationStatus';
+import IdentityVerificationStatus from './IdentityVerificationStatus';
+import MHVTermsAndConditionsStatus from './MHVTermsAndConditionsStatus';
+
+export const AccountSecurityContent = ({
+  isIdentityVerified,
+  isMultifactorEnabled,
+  mhvAccount,
+  showMHVTermsAndConditions,
+  useSSOe,
+}) => {
+  const data = [
+    {
+      title: 'Identity verification',
+      value: (
+        <IdentityVerificationStatus isIdentityVerified={isIdentityVerified} />
+      ),
+    },
+    {
+      title: '2-factor authentication',
+      value: (
+        <TwoFactorAuthorizationStatus
+          isMultifactorEnabled={isMultifactorEnabled}
+          useSSOe={useSSOe}
+        />
+      ),
+    },
+  ];
+
+  if (showMHVTermsAndConditions) {
+    data.push({
+      title: 'Terms and conditions',
+      value: <MHVTermsAndConditionsStatus mhvAccount={mhvAccount} />,
+    });
+  }
```

Comment:
This `data` array looked a little funky to me at first. Instead of constructing the data that way, you could have `ProfileInfoTable` also export a `ProfileInfoTableRow` and do something like -

```jsx
<ProfileInfoTable fieldName="accountSecurity">
    <ProfileInfoTableRow 
        title="Identity verification" 
        value={<IdentityVerificationStatus isIdentityVerified={isIdentityVerified} />
    <ProfileInfoTableRow 
        title="Terms and conditions" 
        value={<MHVTermsAndConditionsStatus mhvAccount={mhvAccount} />
</ProfileInfoTable>
```

It might be overkill for this, but I've been wishing our MegaMenu component was designed with components as building blocks instead of processing a complicated data structure.

Reply:
I definitely like this idea for a future improvement. This seems more React-y. And this would make it easier to add support for different kinds of table rows.